frontend/packages/console-app/src/components/console-operator/ConsoleOperatorConfig.tsx

```diff
@@ -185,101 +245,81 @@ const ConsolePluginsTable: React.FC<ConsolePluginsTableProps> = ({ obj, rows, lo
+                  </Td>
+                  <Td dataLabel={columns[5].id}>
+                    <ConsolePluginCSPStatus hasViolations={hasCSPViolations ?? false} />
+                  </Td>
+                </Tr>
+              ),
+            )}
           </Tbody>
         </Table>
       ) : (
-        <EmptyBox label={t('console-app~console plugins')} />
+        <EmptyBox label={t('console-app~Console plugins')} />
       )}
     </div>
   );
 };
 
 const DevPluginsPage: React.FCC<ConsoleOperatorConfigPageProps> = (props) => {
-  const [pluginInfo, pluginInfoLoaded] = useDynamicPluginInfo();
+  const [pluginInfoEntries, allPluginsProcessed] = useDynamicPluginInfo();
+
   const rows = React.useMemo<ConsolePluginTableRow[]>(
-    () =>
-      !pluginInfoLoaded
-        ? []
-        : pluginInfo.filter(isLoadedDynamicPluginInfo).map((plugin) => ({
-            name: plugin.metadata.name,
-            version: plugin.metadata.version,
-            description: plugin.metadata?.customProperties?.console?.description,
-            enabled: plugin.enabled,
-            status: plugin.status,
-          })),
-    [pluginInfo, pluginInfoLoaded],
+    () => (allPluginsProcessed ? pluginInfoEntries.map(getConsolePluginDisplayData) : []),
+    [pluginInfoEntries, allPluginsProcessed],
   );
-  return <ConsolePluginsTable {...props} rows={rows} loaded={pluginInfoLoaded} />;
+
+  return <ConsolePluginsTable {...props} rows={rows} loaded={allPluginsProcessed} />;
 };
 
 const PluginsPage: React.FC<ConsoleOperatorConfigPageProps> = (props) => {
-  const [pluginInfo, pluginInfoLoaded] = useDynamicPluginInfo();
-  const [consolePlugins, consolePluginsLoaded] = useK8sWatchResource<ConsolePluginKind[]>({
-    isList: true,
-    kind: referenceForModel(ConsolePluginModel),
-  });
+  const [pluginInfoEntries, allPluginsProcessed] = useDynamicPluginInfo();
+
   const enabledPlugins = React.useMemo(() => props?.obj?.spec?.plugins ?? [], [
     props?.obj?.spec?.plugins,
   ]);
+
   const rows = React.useMemo<ConsolePluginTableRow[]>(() => {
-    if (!pluginInfoLoaded || !consolePluginsLoaded) {
+    if (!allPluginsProcessed) {
       return [];
     }
-    return consolePlugins.map((plugin) => {
-      const pluginName = plugin?.metadata?.name;
-      const enabled = enabledPlugins.includes(pluginName);
-      const loadedPluginInfo = pluginInfo
-        .filter(isLoadedDynamicPluginInfo)
-        .find((i) => i?.metadata?.name === pluginName);
-      const notLoadedPluginInfo = pluginInfo
-        .filter(isNotLoadedDynamicPluginInfo)
-        .find((i) => i?.pluginName === pluginName);
-      if (loadedPluginInfo) {
-        return {
-          name: plugin?.metadata?.name,
-          version: loadedPluginInfo?.metadata?.version,
-          description: loadedPluginInfo?.metadata?.customProperties?.console?.description,
-          enabled,
-          status: loadedPluginInfo?.status,
-        };
-      }
+
+    return pluginInfoEntries.map((plugin) => {
```

Comment:
It looks like only `Enabled` plugin will be shown on Console Plugins list page

3 plugins in the cluster but only 2 are shown
https://github.com/user-attachments/assets/15856f78-254b-48af-8a5c-7fb4fe151922

disabled plugin will disappear
https://github.com/user-attachments/assets/a29ef2f7-4896-486a-be1e-2e9f0b213bf6